ciao-launcher/main.go

```diff
@@ -333,6 +331,22 @@ func connectToServer(doneCh chan struct{}, statusCh chan struct{}) {
 		dialCh <- err
 	}()
 
+	var clusterConfig payloads.Configure
+	var data []byte
+	for {
```

Comment:
I don't really like this loop.  It will use loads of CPU and might never exit.  It will also stop launcher from exiting cleanly when there's a SIGTERM.  There must be a better way to access the SSNTP provisioned cluster config.  Is this data guaranteed to be present once dial suceeds?  If so we just need to move this call down a little bit, 
```
		select {
		case err := <-dialCh:
			dialing = false
			if err != nil {
				break DONE
			}

                        < Could we put the call to client.ClusterConfiguration() here?>
```